src/cpp/core/system/PosixSystem.cpp

```diff
@@ -2426,11 +2429,64 @@ Error runProcess(const std::string& path,
    return error;
 }
 
-Error getChildProcesses(std::vector<ProcessInfo> *pOutProcesses, bool populateUsername)
+Error getChildProcesses(
+      std::vector<ProcessInfo>* pOutProcesses,
+      bool populateUsername)
 {
    return getChildProcesses(::getpid(), pOutProcesses, populateUsername);
 }
 
+#ifdef __APPLE__
+
+Error getChildProcesses(
```

Comment:
I found that using `ps` to list child processes was seemingly unreliable on macOS; hence the move to using `libproc` here. ChatGPT was surprisingly helpful!